product images with same name overwrite previous image fix

umarch06 opened the pull request:
### Description
Images were not being renamed while uploading bulk products with images. For example, while importing images from paths a/b.jpg, b/b.jpg, c/b.jpg for three different products were overwriting the same file b.jpg in magento 2. I allowed to rename file if was already existed in Magento 2 catalog product folder. So there is no overwriting while uploading same image name with different folders.

### Manual testing scenarios
1. Create A CSV File for import by same name image within multi directory
2. Upload file in pub/media/import directory with directory
3. Login Backend
4. Got to setting >> import
5. Select "Entity Type" Product
6. Select "Import Behavior" Add/Update
7. Upload CSV File
8. Click Button "Check Data "
9. Click "Import"
10. Import will be successful and it will not override the existing images file with same names.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

dmanners commented on app/code/Magento/CatalogImportExport/Model/Import/Uploader.php:
```diff
@@ -180,6 +180,7 @@ public function move($fileName, $renameFileOff = false)
         $filePath = $this->_directory->getRelativePath($this->getTmpDir() . '/' . $fileName);
         $this->_setUploadFile($filePath);
         $destDir = $this->_directory->getAbsolutePath($this->getDestDir());
+        $this->setAllowRenameFiles(true);
```
I think that it would make more sense to see why the variable `$renameFileOff` is false when passed into the method in this case. It looks like during the lines 152 we are setting the naming options as false.

```
         if ($renameFileOff) {
             $this->setAllowRenameFiles(false);
         }
```

dmanners commented:
Please also take a look at the integration tests for this section of the system.

Commit:
product images while exporting does not override previous image having same name in other folder

umarch06 commented:
@dmanners, probably I missed seeing the `if` check, that is fixed in new commit.

dmanners committed:
magento-engcom/import-export-improvements#50: exclude the image attributes from the product test,
 - this is done as the image paths are now being rewritten and are already covered in the import/export tests

dmanners committed:
magento-engcom/import-export-improvements#50: use the assertEqualsSpecificAttributes to check images contain the same name
 - when importing files with the same name will now be renamed for this reason we check names are similar now

magento-engcom-team commented:
@umarch06 thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.
